report/colles/classes/report.php

```diff
@@ -186,10 +186,39 @@ public function has_student_report() {
      */
     public function has_childreports($canaccessreports) {
         if ($canaccessreports) {
+            $questionreports = array();
+            $key = get_string('fieldset_content_01', 'surveyprotemplate_'.$this->surveypro->template);
```

Comment:
I'm afraid but I don't like lang strings being used as keys in arrays. Instead you should use here the proper keys ('fieldset_content_02', 'summary', ....) and only at the end, in `surveypro_add_report_link()` perform the calls to `get_string()` and use the language strings.

Also, while it's nice to have all these in the navigation block... is the report navigable without using the navigation block? Sort of tab on top or so?